allennlp/training/log_writer.py

```diff
@@ -0,0 +1,269 @@
+from typing import Any, Callable, Dict, List, Optional, Set
+import logging
+
+import torch
+
+from allennlp.common.from_params import FromParams
+from allennlp.data import TensorDict
+from allennlp.nn import util as nn_util
+from allennlp.training.optimizers import Optimizer
+from allennlp.training import util as training_util
+from allennlp.models.model import Model
+
+logger = logging.getLogger(__name__)
+
+
+class LogWriter(FromParams):
+    """
+    Class that handles trainer logging.
+
+    # Parameters
+
+    summary_interval : `int`, optional (default = `100`)
+        Most statistics will be written out only every this many batches.
+    distribution_interval : `int`, optional (default = `None`)
+        If provided, activation distributions will be written out every this many batches.
+        If None, activation distributions will not be written out.
+        When this parameter is specified, the following additional logging is enabled:
+            * Distributions of model parameters
+            * The ratio of parameter update norm to parameter norm
+            * Distribution of layer activations
```

Comment:
This isn't rendering correctly in the docs. May need blank lines before and after this list.
```suggestion

            * Distributions of model parameters
            * The ratio of parameter update norm to parameter norm
            * Distribution of layer activations
            
```